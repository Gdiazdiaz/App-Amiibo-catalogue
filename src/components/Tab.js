function Tab(props) {
    const data = props;
    let dataID = '';
    let num = data.index%3;
    if(data.index%3===0){
        console.log(data.index%3);
        dataID = 'color1';
    }
    else {
        console.log(data.index%3);
        dataID = 'color2';
    }
    console.log(num);
    return(
        <div className="card-container"  id={dataID}>
            <img className="card-pic" src={data.imageUrl}/>
            <div className="details-container">
                <h3 className="card-name">{data.name}</h3>
                <h4 className="card-filnumb">Movies: {data.films}</h4>
            </div>
        </div>
    );
}

export default Tab;