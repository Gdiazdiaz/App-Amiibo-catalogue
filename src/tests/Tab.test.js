import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Tab from '../components/Tab';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';

describe('Testing page deployment', () => {
  test('Profile components matches snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Tab />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
