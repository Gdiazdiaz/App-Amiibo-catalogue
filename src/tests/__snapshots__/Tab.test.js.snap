// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing page deployment Profile components matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-container"
      >
        <img
          class="card-pic"
        />
        <div
          class="details-container"
        >
          <h3
            class="card-name"
          />
          <h4
            class="card-filnumb"
          >
            Movies:
          </h4>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-container"
    >
      <img
        class="card-pic"
      />
      <div
        class="details-container"
      >
        <h3
          class="card-name"
        />
        <h4
          class="card-filnumb"
        >
          Movies:
        </h4>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
